docs/Configuration/fleet-server-configuration.md

```diff
@@ -213,6 +241,8 @@ By default, this will set up a Redis pool for that configuration and execute a
 
 For the address of the Redis server that Fleet should connect to, include the hostname and port.
 
+If an AWS ElastiCache endpoint is specified and `redis_password` isn't specified, Identity and Access Management (IAM) authentication is automatically used.
```

Comment:
```suggestion
If an AWS ElastiCache endpoint is specified and `redis_password` isn't specified, Fleet will first attempt to use Identity and Access Management (IAM) authentication before falling back to standard auth without a password.
```

@noahtalerman This relates to your question below about roles as well. If we don't want to add an explicit `redis_use_iam` flag, the only way we can support IAM-less connections without a password is to first _try_ to get a password via IAM, and if that fails, try to connect without a password.

For the `redis_sts_assume_role_arn` config, I would say something like:

> Amazon Resource Name (ARN) of the AWS Security Token Service (STS) role to assume when using Identity and Access Management (IAM) for Redis authentication.